src/lib/components/chat/ChatInput.svelte

```diff
@@ -35,6 +36,23 @@
 			textareaElement.focus();
 		}
 	});
+
+	$: onMount(() => {
+		const handlePaste = (event: ClipboardEvent) => {
+			if (!textareaElement.contains(document.activeElement)) return;
+
+			const items = event.clipboardData?.items;
+			if (!items) return;
+
+			for (const item of items) {
+				if (item.kind !== "file") continue;
+				const file = item.getAsFile();
+				if (file) files = [...files, file];
+			}
+		};
+		document.addEventListener("paste", handlePaste);
+		return () => document.removeEventListener("paste", handlePaste);
+	});
```

Comment:
```suggestion
		function handlePaste (event: ClipboardEvent) => {
			if (!textareaElement.contains(document.activeElement)) return;

			const items = event.clipboardData?.items;
			if (!items) return;

			for (const item of items) {
				if (item.kind !== "file") continue;
				const file = item.getAsFile();
				if (file) files = [...files, file];
			}
		};
```

Then forward the event from ChatInput with `on:paste` and just use <ChatInput on:paste={handlePaste} />

Reply:
https://github.com/huggingface/chat-ui/pull/1212